packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -2265,7 +2285,10 @@ found '${rule.provider}' assigned.`,
     };
 
     // Get and validate the auth rules.
-    return getArg('rules', []) as AuthRule[];
+    const authRules = getArg('rules', []) as AuthRule[];
+
+    // All the IAM auth rules that are added using @auth directive need IAM policy to be generated. AuthRules added for AdminUI don't
+    return authRules.map(rule => (rule.provider === 'iam' ? { ...rule, generateIAMPolicy: true } : rule));
```

Comment:
Should we have this set to assume the transformer is creating the IAM Policy unless specified?